### PR TITLE
feat: add confirm signup method

### PR DESCRIPTION
**Pull request by jkdowdle**
## Motivation

~based on #83~

We want to provide a way to sign-up Platform users easily, especially in 3rd party apps like Preventia.

### CODE REVIEW

File: src/APIBasedAuth.ts
Comment (swain) [inline src/APIBasedAuth.ts:246]: I think we should call `this.setTokens(...)` before this return, similar to line 202

**Review by aecorredor — COMMENTED** [review]
echoing Swain's comment.

**Comment by aecorredor** [comment]
@jkdowdle btw, I noticed this repo is not enforcing checks before merging. We should probably enable those in the GitHub settings for this repo.

File: src/APIBasedAuth.ts
Comment (jkdowdle) [reply src/APIBasedAuth.ts:246]: 🤦

**Comment by jkdowdle** [comment]
> @jkdowdle btw, I noticed this repo is not enforcing approvals before merging. We should probably enable those in the GitHub settings for this repo.

do you still see this? I wonder if it was because before this pr was referencing a branch other than master

**Comment by loscm** [comment]
Requested review from a random `@lifeomic/security` team member, @bishopb, due to term: **token**.

This review is not blocking and is for broader awareness.  Consider if this change requires deeper security review and ask when it is necessary.

- [Automation](https://github.com/lifeomic/pr-security-flag-scanner/blob/master/src/lambdas/pr-security-flag-scanner/index.ts) | [#pr-flagged-sec-review](https://lifeomic.slack.com/messages/pr-flagged-sec-review)

**Review by rob-snyder — APPROVED** [review]
LGTM

**Comment by aecorredor** [comment]
> > @jkdowdle btw, I noticed this repo is not enforcing approvals before merging. We should probably enable those in the GitHub settings for this repo.
> 
> do you still see this? I wonder if it was because before this pr was referencing a branch other than master

Maybe it was that. I can't tell now since it already had Rob's approval.

**Review by aecorredor — APPROVED** [review]

**Comment by rob-snyder** [comment]
> > > @jkdowdle btw, I noticed this repo is not enforcing approvals before merging. We should probably enable those in the GitHub settings for this repo.
> > 
> > 
> > do you still see this? I wonder if it was because before this pr was referencing a branch other than master
> 
> Maybe it was that. I can't tell now since it already had Rob's approval.

I saw it as blocked waiting for review prior to my approval, so I think it was the fact that it was referencing a different branch

**Pull request merged**
